tasks/eth/006-MCP-L1/SignFromJson.s.sol

```diff
@@ -0,0 +1,346 @@
+        require(oracleToCheck.l2BlockTime() == l2BlockTime, "3900");
+
+        require(oracleToCheck.PROPOSER() == l2OutputOracleProposer, "4000");
+        require(oracleToCheck.proposer() == l2OutputOracleProposer, "4100");
+
+        require(oracleToCheck.CHALLENGER() == l2OutputOracleChallenger, "4200");
+        require(oracleToCheck.challenger() == l2OutputOracleChallenger, "4300");
+
+        require(oracleToCheck.FINALIZATION_PERIOD_SECONDS() == finalizationPeriodSeconds, "4400");
+        require(oracleToCheck.finalizationPeriodSeconds() == finalizationPeriodSeconds, "4500");
+
+        require(oracleToCheck.startingBlockNumber() == l2OutputOracleStartingBlockNumber, "4600");
+        require(oracleToCheck.startingTimestamp() == l2OutputOracleStartingTimestamp, "4700");
+    }
+
+    /// @notice Asserts that the OptimismMintableERC20Factory is setup correctly
+    function checkOptimismMintableERC20Factory() internal view {
+        console.log("Running chain assertions on the OptimismMintableERC20Factory");
+
+        require(proxies.OptimismMintableERC20Factory.code.length != 0, "4800");
+
+        OptimismMintableERC20Factory factoryToCheck = OptimismMintableERC20Factory(proxies.OptimismMintableERC20Factory);
+
+        require(factoryToCheck.BRIDGE() == proxies.L1StandardBridge, "4900");
+        require(factoryToCheck.bridge() == proxies.L1StandardBridge, "5000");
+        require(factoryToCheck.bridge().code.length != 0, "5001");
+    }
+
+    /// @notice Asserts that the L1ERC721Bridge is setup correctly
+    function checkL1ERC721Bridge() internal view {
+        console.log("Running chain assertions on the L1ERC721Bridge");
+
+        require(proxies.L1ERC721Bridge.code.length != 0, "5100");
+
+        L1ERC721Bridge bridgeToCheck = L1ERC721Bridge(proxies.L1ERC721Bridge);
+
+        require(address(bridgeToCheck.OTHER_BRIDGE()) == Predeploys.L2_ERC721_BRIDGE, "5200");
+        require(address(bridgeToCheck.otherBridge()) == Predeploys.L2_ERC721_BRIDGE, "5300");
+        require(address(bridgeToCheck.otherBridge()).code.length != 0, "5301");
+
+        require(address(bridgeToCheck.MESSENGER()) == proxies.L1CrossDomainMessenger, "5400");
+        require(address(bridgeToCheck.messenger()) == proxies.L1CrossDomainMessenger, "5500");
+        require(address(bridgeToCheck.messenger()).code.length != 0, "5502");
+
+        require(address(bridgeToCheck.superchainConfig()) == proxies.SuperchainConfig, "5600");
+        require(address(bridgeToCheck.superchainConfig()).code.length != 0, "5603");
+    }
+
+    /// @notice Asserts the OptimismPortal is setup correctly
+    function checkOptimismPortal() internal view {
+        console.log("Running chain assertions on the OptimismPortal");
+
+        require(proxies.OptimismPortal.code.length != 0, "5700");
+
+        OptimismPortal portalToCheck = OptimismPortal(payable(proxies.OptimismPortal));
+
+        address guardian = superchainConfigGuardian;
```

Comment:
smol style nit: doing extra assignment here while not doing this in other functions. Not the end of the world, just calling out how consistent style is important